lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -412,6 +415,136 @@ def merge_elements(self, document: Document) -> Document:
+    Example:
+         .. code-block:: python
+
+            llm = OpenAI(OpenAIModels.GPT_4O, api_key = '')
+
+            prompt = "Analyze two CSV tables that may be parts of a single table split across pages. Determine if the second table is a continuation of the first with 100% certainty. Check either of the following:\
+            1. Column headers: Must be near identical in terms of text(the ordering/text may contain minor errors because of OCR quality) in both tables. If the headers are almost the same check the number of columns, they should be roughly the same.\
+            2. Missing headers: If the header/columns in the second table are missing, then the first row in the second table should logically be in continutaion of the last row in the first table.\
+            Respond with only 'true' or 'false' based on your certainty that the second table is a continuation. Certainty is determinedx if either of the two conditions is true."
+
+            regex_pattern = r"table \d+"
+
+            merger = TableMerger(llm_prompt = prompt, llm=llm)
+
+            context = sycamore.init()
+            pdf_docset = context.read.binary(paths, binary_format="pdf", regex_pattern= regex_pattern)
+                .partition(partitioner=ArynPartitioner())
+                .merge(merger=merger)
+    """
+
+    def __init__(self, regex_pattern=None, llm_prompt=None, llm=None, *args, **kwargs):
+        self.regex_pattern = regex_pattern
+        self.llm_prompt = llm_prompt
+        self.llm = llm
+
+    def merge_elements(self, document: Document) -> Document:
+
+        table_elements = [ele for ele in document.elements if ele.type == "table"]
+        if len(table_elements) < 1:
+            return document
+        if self.regex_pattern:
+            document.elements = self.customTableHeaderAdditionFilter(document.elements)
+        if not self.llm_prompt or len(table_elements) < 2:
+            return document
+        document = self.process_llm_query(document)
+        table_elements = [ele for ele in document.elements if ele.type == "table"]
+        other_elements = [ele for ele in document.elements if ele.type != "table"]
+        new_table_elements = [table_elements[0]]
+        for element in table_elements[1:]:
+            if self.should_merge(new_table_elements[-1], element):
+                new_table_elements[-1] = self.merge(new_table_elements[-1], element)
+            else:
+                new_table_elements.append(element)
+        other_elements.extend(new_table_elements)
+        document.elements = other_elements
+        return document
+
+    def should_merge(self, element1: Element, element2: Element) -> bool:
+        if "true" in element2["properties"]["table_continuation"].lower():
+            return True
+        return False
+
+    def merge(self, elt1: Element, elt2: Element) -> Element:
+
+        new_elt = Element()
+        new_elt.type = "table"
```

Comment:
also prob need to find a way to merge the cells in the underlying table objects.
something like
```
translate cells in table2 down by table1.nrows rows
concat the cell lists and construct a new table
```

Reply:
done